server/models/ConnectedAccount.js

```diff
@@ -72,5 +75,15 @@ export default (Sequelize, DataTypes) => {
     });
   };
 
+  ConnectedAccount.prototype.delete = async function(remoteUser) {
+    mustBeLoggedInTo(remoteUser, 'disconnect this connected account');
+    if (remoteUser.id !== this.CreatedByUserId || !remoteUser.isAdmin(this.CollectiveId)) {
+      throw new errors.Unauthorized({
+        message: 'You are either logged out or not authorized to disconnect this account',
+      });
+    }
+    return this.destroy();
+  };
+
```

Comment:
It seems to me that checking authentication/permissions and generating GraphQL errors is not the role of the model (whose role should be to solely map the data model to the database). Though I don't know this part very well, this code could be problematic for example in the Forest admin interface where we may call `destroy` as an admin action without providing a `remoteUser`.

I'll add that it makes the API for it pretty unclear, when reading at this line above it's hard to tell why we provide the remoteUser to the function at first glance.

```es6
await Account.delete(remoteUser);
```

Note that this pattern has been followed in other parts of the code in the past so it's not a big issue but as a design principle I'd be in favor of moving the permission check in the controllers/graphql endpoints for any new development (in this case `server/controllers/connectedAccounts.js`)

Reply:
Please review again, I have refactored the authorization part.